shell_command returns freeform output

pakrym-oai opened the pull request:
Instead of returning structured out and then re-formatting it into freeform, return the freeform output from shell_command tool.

Keep `shell` as the default tool for GPT-5.

pakrym-oai committed:
Make shell_command return freeform output

pakrym-oai committed:
tests: simplify tool expectations and remove unused imports

pakrym-oai committed:
tests: remove o3 model tool assertion and update gpt-5 toolset expectations

- Dropped the o3 model tools assertion from model_selects_expected_tools.
- Added "apply_patch" as an expected tool for gpt-5-codex.
- Clarified test description for gpt-5 toolset assertion.

pakrym-oai committed:
core: only show "Total output lines" when it differs from formatted output count

tests: remove ShellModelOutput::ShellCommand cases to match updated model output formatting

dylan-hurd-oai commented on codex-rs/core/src/tools/mod.rs:
```diff
@@ -61,6 +62,33 @@ pub fn format_exec_output_for_model(exec_output: &ExecToolCallOutput) -> String
+        &exec_output.aggregated_output.text,
+        total_lines,
+        TELEMETRY_PREVIEW_MAX_BYTES,
+        TELEMETRY_PREVIEW_MAX_LINES,
```
Codex seems right on this one?

pakrym-oai committed:
tools: reuse formatter for exec output and update limits

Introduce format_exec_output_for_model helper on ToolEmitter to select
freeform vs structured formatting consistently. Use it for success and
sandbox timeout/denied paths. Update freeform exec output truncation to
use SHELL_OUTPUT_MAX_BYTES and a fixed 256 line limit.

pakrym-oai committed:
tests: add coverage for shell command output truncation around 10k bytes

pakrym-oai committed:
tests/shell_serialization: use DangerFullAccess policy instead of ReadOnly for shell_command output tests

pakrym-oai committed:
Merge branch 'main' into pakrym/make-shell-command-return-freeform-output

# Conflicts:
#	codex-rs/core/tests/suite/prompt_caching.rs
